heap/max_heap.py

```diff
@@ -10,8 +10,19 @@ def insert(self, value):
 
 
   def delete(self):
-      self.storage.pop(0)
-      return self._bubble_up(len(self.storage) -1)
+      last_index =  self.storage[len(self.storage) - 1]
```

Comment:
this will seem a bit nitpicky, but it's to help others (and yourself) to have a clear idea of what's going on:
your naming this `last_index`, but you're not assigning and index - you're assigning a value (`self.storage[_some_index_here]`).
I suggest you either change the name to `last_value`, or you actually pass in an index `last_index = len(self.storage) - 1`.
if `last_index` is actually and index, you should be able to use it like so
`self.storage[last_index]`